src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -2086,8 +2086,14 @@ void GameObject::CastSpell(Unit* target, uint32 spellId)
         if (owner->HasUnitFlag(UNIT_FLAG_PLAYER_CONTROLLED))
             trigger->SetUnitFlag(UNIT_FLAG_PLAYER_CONTROLLED);
         if (owner->IsFFAPvP())
-            trigger->SetByteFlag(UNIT_FIELD_BYTES_2, 1, UNIT_BYTE2_FLAG_FFA_PVP);
+        {
+            if (!HasByteFlag(UNIT_FIELD_BYTES_2, 1, UNIT_BYTE2_FLAG_FFA_PVP))
+            {
+                sScriptMgr->OnFfaPvpStateUpdate(trigger, true);
+                trigger->SetByteFlag(UNIT_FIELD_BYTES_2, 1, UNIT_BYTE2_FLAG_FFA_PVP);
+            }
```

Comment:
From the crashlog, this is related. I found it odd too :(

Reply:
Thread #19.2  is calling GameObject line 2092. 2092 is the event handler. Could it be something in an event payload in a lua script?
Should Event caLLs be wrapped in a try / catch / swallow to avoid crashing the server if the event overloads Blow out?

Reply:
I'm pretty sure there is no lua script running with that hook yet, although It's pretty odd indeed. It was a frost trap that made it crash, which is shown on the crashlog, but I don't have more information about it sadly :/ 
The event per-se it's okay, it doesn't need a try catch

Reply:
Seems to me the biggest change is the 
`    ASSERT(offset < 4);`
That HasByte Performs compared to the 

```
    if (offset > 3)
    {
        LOG_ERROR("entities.object", "Object::SetByteFlag: wrong offset {}", offset);
        return;
    }
```
That SetByteFlag Does

Anyone guess why HasByteFlag needs to validate that offset <4?

Reply:
https://github.com/azerothcore/mod-eluna/pull/63 went in a few hours back not sure how quick the server picks those up

Reply:
While the mod-Eluna PR is merged, there are no events registered yet in any Lua scripts.